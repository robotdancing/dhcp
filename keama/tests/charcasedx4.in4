# lcase/ucase data expression

# authoritative is mandatory
authoritative;

# empty configs are not accepted by Kea
default-lease-time 1800;

# reduce literals
class "literal" {
    match if option host-name = concat(ucase("www."), "example.", lcase("COM"));
}
