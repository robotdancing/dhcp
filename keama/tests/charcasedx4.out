{
  # lcase/ucase data expression
  # authoritative is mandatory
  "Dhcp4": {
    "valid-lifetime": 1800,
    "client-classes": [
      # reduce literals
      {
        "name": "literal",
        /// from: match if (option dhcp.host-name) = (concat(uppercase('www.'), concat('example.', lowercase('COM'))))
        "test": "option[12].hex == 'WWW.example.com'"
      }
    ]
  }
}
